test/benchmark/main.cc

```diff
@@ -1,16 +1,40 @@
 // NOLINT(namespace-envoy)
 // This is an Envoy driver for benchmarks.
+#include "test/benchmark/main.h"
+
 #include "test/test_common/environment.h"
 
 #include "benchmark/benchmark.h"
+#include "tclap/CmdLine.h"
+
+static bool skip_expensive_benchmarks = false;
 
-// Boilerplate main(), which discovers benchmarks and runs them.
+// Boilerplate main(), which discovers benchmarks and runs them. This uses two
+// different flag parsers, so the order of flags matters: flags defined here
+// must be passed first, and flags defined in benchmark::Initialize second,
+// separated by --.
+// TODO(pgenera): convert this to abseil/flags/ when benchmark also adopts abseil.
 int main(int argc, char** argv) {
   Envoy::TestEnvironment::initializeTestMain(argv[0]);
 
-  benchmark::Initialize(&argc, argv);
-  if (benchmark::ReportUnrecognizedArguments(argc, argv)) {
-    return 1;
+  // NOLINTNEXTLINE(clang-analyzer-optin.cplusplus.VirtualCall)
+  TCLAP::CmdLine cmd("envoy-benchmark-test", ' ', "0.1");
+  TCLAP::SwitchArg skip_switch("s", "skip_expensive_benchmarks",
+                               "skip or minimize expensive benchmarks", cmd, false);
+
+  cmd.setExceptionHandling(false);
+  try {
+    cmd.parse(argc, argv);
+  } catch (const TCLAP::ExitException& e) {
+    // parse() throws an ExitException with status 0 after printing the output
+    // for --help and --version.
+    return 0;
   }
+
+  skip_expensive_benchmarks = skip_switch.getValue();
+
+  benchmark::Initialize(&argc, argv);
   benchmark::RunSpecifiedBenchmarks();
 }
+
+bool SkipExpensiveBenchmarks() { return skip_expensive_benchmarks; }
```

Comment:
s/SkipExpensiveBenchmarks/skipExpensiveBenchmarks/g to fix CI (naming convention).